src/java.base/share/classes/java/net/SocketOptions.java

```diff
@@ -190,6 +190,14 @@ public interface SocketOptions {
 
     /**
      * See {@link StandardSocketOptions#SO_LINGER} for description of this socket option.
+     * <p>
+     * Either {@linkplain Boolean#FALSE Boolean.FALSE} or an integer value lesser than {@code 0}
+     * can be passed to {@link #setOption(int, Object)} to disable this option. An integer
+     * value {@code >= 0} passed to {@code setOption(int, Object)} will enable the option and
+     * the value will represent the linger timeout.
```

Comment:
I think the wording could be improved a bit with "Set the value to Boolean.FALSE or an integer less than 0 with setOption to disable this option".

Reply:
Done. I've updated the PR with that suggested wording. I decided to let the part which explains that an integer >=0 enables this option stay. I hope that's OK.